### PR TITLE
Added query guidance for M:M in JS

### PR DESCRIPTION
**Pull request by svidgen**
*Issue #, if available:*

https://github.com/aws-amplify/amplify-js/issues/8139

*Description of changes:*

The correct way to query many-to-many relationships isn't completely intuitive. This PR adds guidance.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/lib/datastore/fragments/js/relational/query-many-snippet.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+).map(pe => pe.post);
+
+// All editors on a given post
+const editorsByPost = (await DataStore.query(PostEditor)).filter(
+    pe => pe.post.id === post.id
+).map(pe => pe.editor);
+
+// All editors of posts where title contains "first"
+const editorsOfFirstPosts = (await DataStore.query(PostEditor)).filter(
+    pe => pe.post.title.match(/first/i)
```
Comment (iartemiev) [inline docs/lib/datastore/fragments/js/relational/query-many-snippet.md:41]: Let's use `String.includes` here, regex may be a bit cryptic for newer devs

E.g., `pe => pe.post.title.includes('first')`

**Review by iartemiev — CHANGES_REQUESTED** [review]
Looks good, just 1 minot suggestions

File: docs/lib/datastore/fragments/js/relational/query-many-snippet.md (hunk 1 above)
Comment (svidgen) [reply docs/lib/datastore/fragments/js/relational/query-many-snippet.md]: Updated!

**Review by iartemiev — APPROVED** [review]

**Review by renebrandel — COMMENTED** [review]
Suggestions

**Review by renebrandel — APPROVED** [review]

**Review by iartemiev — APPROVED** [review]

**Pull request merged**